XamarinCommunityToolkitSample/Pages/Behaviors/CharactersValidationBehaviorPage.xaml

```diff
@@ -0,0 +1,70 @@
+            <Setter Property="TextColor"
+                    Value="Red" />
+        </Style>
+    </pages:BasePage.Resources>
+
+    <StackLayout Padding="{StaticResource ContentPadding}"
+                 Spacing="25"
+                 VerticalOptions="StartAndExpand">
+        <Label Text="Text color will change accordingly to the style that is configured when an invalid string is entered." />
+        <Frame Margin="{StaticResource ContentPadding}"
+               CornerRadius="10">
+            <Grid>
+                <Grid.RowDefinitions>
+                    <RowDefinition Height="Auto" />
+                    <RowDefinition Height="Auto" />
+                    <RowDefinition Height="Auto" />
+                </Grid.RowDefinitions>
+                <Grid.ColumnDefinitions>
+                    <ColumnDefinition Width="*" />
+                    <ColumnDefinition Width="*" />
+                </Grid.ColumnDefinitions>
+                <Label Text="CharacterType" />
+                <Picker x:Name="CharacterTypePicker"
+                        Title="CharacterType"
+                        Grid.Column="1"
+                        ItemsSource="{Binding CharacterTypes,
+                                              Source={x:Reference Page}}"
+                        SelectedIndex="1" />
+                <Label Grid.Row="1"
+                       Text="MinimumCharacterCountEntry" />
+                <Entry x:Name="MinimumCharacterCountEntry"
+                       Grid.Row="1"
+                       Grid.Column="1"
+                       Keyboard="Numeric"
+                       Text="2" />
+                <Label Grid.Row="2"
+                       Text="MaximumCharacterCountEntry" />
+                <Entry x:Name="MaximumCharacterCountEntry"
+                       Grid.Row="2"
+                       Grid.Column="1"
+                       Keyboard="Numeric"
+                       Text="20" />
+            </Grid>
+        </Frame>
+        <Label Text="Type characters for validation behavior according to the settings you set upon." />
+        <Entry Placeholder="Type characters...">
+            <Entry.Behaviors>
+                <behaviors:CharactersValidationBehavior CharacterType="{Binding SelectedItem,
+                                                                                Source={x:Reference CharacterTypePicker}}"
+                                                        InvalidStyle="{StaticResource InvalidEntryStyle}"
+                                                        MaximumCharacterCount="{Binding Text,
+                                                                                        Source={x:Reference MaximumCharacterCountEntry}}"
+                                                        MinimumCharacterCount="{Binding Text,
+                                                                                        Source={x:Reference MinimumCharacterCountEntry}}" />
```

Comment:
Shouldn't we use StringToIntegerConverter or smth like that?

Reply:
the StringToIntegerConverter it's not a great idea, it raised an ArgumentNullException that make the App crash (try tapping a point with the numerical keaybord).
Maybe with a safer one (sending null instead of an Exception and using TargetNullValue to give a default value) ?
Maybe adding a validation behavior too for the settings Entry ?